src/nm-config-data.c

```diff
@@ -674,7 +674,7 @@ nm_config_data_log (const NMConfigData *self,
 		is_atomic = nm_config_data_is_intern_atomic_group (self, group);
 
 		_LOG (stream, prefix, "");
-		_LOG (stream, prefix, "[%s]%s", group, is_atomic && !stream ? " # atomic section" : "");
+		_LOG (stream, prefix, "[%s]%s", group, is_atomic ? " # atomic section" : "");
```

Comment:
Do you know why does cppcheck complain about line 666? This doesn't make sense to me.

Anyway, it seems that glib's keyfile parser does not actually allow comments after `[group]`. While this is only for logging, we should log valid keyfile format, hence this comment should move to the following line, like
```C
_LOG (stream, prefix, "[%s]", group);
if (is_atomic)
    _LOG (stream, prefix, "# atomic section");
```

Reply:
I need to have a deeper look at `_LOG` macro. probably, I'll drop this commit for now and reopen it later